modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -99,31 +93,33 @@ public void startTrack() {
         inBusyLock(busyLock, () -> {
             msManager.registerPrefixWatch(PLACEMENTDRIVER_LEASES_KEY, updateListener);
 
-            CompletableFuture<VaultEntry> entryFut = vaultManager.get(PLACEMENTDRIVER_LEASES_KEY);
-
-            VaultEntry entry = entryFut.join();
+            msManager.recoveryFinishedFuture().thenAccept(recoveryRevision -> {
+                Entry entry = msManager.getLocally(PLACEMENTDRIVER_LEASES_KEY, recoveryRevision);
```

Comment:
The same as above. Invokations of two method of one object in a row look unwell.

Reply:
Same answer as above. Let's continue in one topic only.